okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -402,247 +691,33 @@ private static TenantModuleDescriptor getNextTM(Map<String, ModuleDescriptor> mo
-    List<String> errors = new LinkedList<>();
-    while (it.hasNext()) {
-      String runningModule = it.next();
-      ModuleDescriptor rm = modsEnabled.get(runningModule);
-      if (md.getProduct().equals(rm.getProduct())) {
-        logger.info("resolveModuleConflicts from {}", runningModule);
-        for (TenantModuleDescriptor tm : tml) {
-          if (tm.getId().equals(rm.getId()) && Boolean.TRUE.equals(tm.getFixed())) {
-            errors.add(messages.getMessage("10209", rm.getId()));
-            return errors;
-          }
-        }
-        it.remove();
-        fromModule.add(rm);
-      } else {
-        for (InterfaceDescriptor mi : md.getProvidesList()) {
-          for (InterfaceDescriptor pi : rm.getProvidesList()) {
-            if (pi.isRegularHandler()) {
-              String confl = pi.getId();
-              if (mi.getId().equals(confl)
-                  && mi.isRegularHandler()
-                  && modsEnabled.containsKey(runningModule)) {
-                logger.info("resolveModuleConflicts remove {}", runningModule);
-                TenantModuleDescriptor tm = new TenantModuleDescriptor();
-                tm.setAction(TenantModuleDescriptor.Action.disable);
-                tm.setId(runningModule);
-                tml.add(tm);
-                it.remove();
-              }
-            }
-          }
-        }
-      }
-    }
-    return errors;
-  }
-
-  private static void addOrReplace(List<TenantModuleDescriptor> tml, ModuleDescriptor md,
-                                   TenantModuleDescriptor.Action action,
-                                   ModuleDescriptor fm) {
-    logger.info("addOrReplace from {} to id {}", fm != null ? fm.getId() : "null", md.getId());
-    Iterator<TenantModuleDescriptor> it = tml.iterator();
-    boolean found = false;
-    boolean fixed = false;
-    while (it.hasNext()) {
-      TenantModuleDescriptor tm = it.next();
-      if (tm.getAction().equals(action) && tm.getId().equals(md.getId())) {
-        fixed = tm.getFixed();
-        it.remove();
-      } else if (fm != null && tm.getAction() == TenantModuleDescriptor.Action.enable
-          && tm.getId().equals(fm.getId())) {
-        logger.info("addOrReplace .. patch id {}", md.getId());
-        tm.setId(md.getId());
-        found = true;
-      }
-    }
-    if (found) {
-      return;
-    }
-    TenantModuleDescriptor t = new TenantModuleDescriptor();
-    t.setAction(action);
-    t.setFixed(fixed);
-    t.setId(md.getId());
-    if (fm != null) {
-      t.setFrom(fm.getId());
-    }
-    tml.add(t);
-  }
-
-  private static void upgradeLeafs(
-      Map<String, ModuleDescriptor> modsAvailable,
-      Map<String, ModuleDescriptor> modsEnabled, List<TenantModuleDescriptor> tml) {
-    while (upgradeLeafs2(modsAvailable, modsEnabled, tml)) {
-      // something upgraded.. try again
-    }
-  }
+  static Map<String, ModuleDescriptor> checkInterfaceProvAvailable(
```

Comment:
Can you add javadoc about the return value?

Reply:
sure